project/cmake/CMakeLists.txt

```diff
@@ -7,7 +7,7 @@ add_subdirectory (test)
 
 if(NOT EMSCRIPTEN)
 
-find_package(Qt5 QUIET)
+find_package(Qt5 COMPONENTS Widgets Gui Core QUIET)
```

Comment:
The component selection will not work if Qt is not installed (hard error).

Reply:
I will revert this for now, but we should find a solution works for both cases, because previous solution fails to build the qtest project:
```
/home/simon/Desktop/vizzu/vizzu-lib/project/cmake/CMakeLists.txt(10):  find_package(Qt5 QUIET )
/usr/lib/x86_64-linux-gnu/cmake/Qt5/Qt5ConfigVersion.cmake(2):  set(PACKAGE_VERSION 5.12.8 )
/usr/lib/x86_64-linux-gnu/cmake/Qt5/Qt5ConfigVersion.cmake(4):  if(PACKAGE_VERSION VERSION_LESS PACKAGE_FIND_VERSION )
/usr/lib/x86_64-linux-gnu/cmake/Qt5/Qt5ConfigVersion.cmake(6):  else()
/usr/lib/x86_64-linux-gnu/cmake/Qt5/Qt5ConfigVersion.cmake(7):  set(PACKAGE_VERSION_COMPATIBLE TRUE )
/usr/lib/x86_64-linux-gnu/cmake/Qt5/Qt5ConfigVersion.cmake(8):  if(PACKAGE_FIND_VERSION STREQUAL PACKAGE_VERSION )
/usr/lib/x86_64-linux-gnu/cmake/Qt5/Qt5Config.cmake(2):  if(CMAKE_VERSION VERSION_LESS 3.1.0 )
/usr/lib/x86_64-linux-gnu/cmake/Qt5/Qt5Config.cmake(6):  if(NOT Qt5_FIND_COMPONENTS )
/usr/lib/x86_64-linux-gnu/cmake/Qt5/Qt5Config.cmake(7):  set(Qt5_NOT_FOUND_MESSAGE The Qt5 package requires at least one component )
/usr/lib/x86_64-linux-gnu/cmake/Qt5/Qt5Config.cmake(8):  set(Qt5_FOUND False )
/usr/lib/x86_64-linux-gnu/cmake/Qt5/Qt5Config.cmake(9):  return()
```

Reply:
I think the `module mode` can be the solution, but i'm not a qt cmake expert:
https://cmake.org/cmake/help/latest/command/find_package.html#search-modes

Reply:
I will check only the qt core comonent, hopefully it will work
```

find_package(Qt5Core QUIET)
if( Qt5Core_FOUND )
    add_subdirectory (qtest)
endif()
```

Reply:
Nice solution in last commit, I resolve this CR